src/emu/softlist.cpp

```diff
@@ -1156,7 +1140,10 @@ void softlist_parser::parse_data_start(const char *tagname, const char **attribu
 			else if (loadflag != nullptr && strcmp(loadflag, "continue") == 0)
 				add_rom_entry(nullptr, nullptr, offset, length, ROMENTRYTYPE_CONTINUE | ROM_INHERITFLAGS);
 			else if (loadflag != nullptr && strcmp(loadflag, "fill") == 0)
-				add_rom_entry(nullptr, (const char *)(FPTR)(strtol(value, nullptr, 0) & 0xff), offset, length, ROMENTRYTYPE_FILL);
+			{
+				auto fill_string = string_format("%02X", strtol(value, nullptr, 0) & 0xff);
```

Comment:
This is one of those cases where `auto` is really unwarranted and doesn't help - the desired type is known, it isn't some complicated inferred iterator or anything.  Please don't use auto for things like this.

Also, I'm pretty sure this is not what you want to do.  The code previously attempted to parse the value as a string, convert to an integer, cast up to a pointer and then store that.  You're not doing that at all, you're storing the `c_str()` address of a temporary variable which is going to be some unpredictable stack address given how short the string is.